### PR TITLE
Add MMR Migration

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

File: frame/header-mmr/src/lib.rs [hunk 1]
```diff
@@ -105,10 +105,38 @@ pub mod pallet {
...
 
+	/// Migration step.
+	#[pallet::storage]
+	#[pallet::getter(fn migration_step)]
+	pub type MigrationStep<T> = StorageValue<_, u32, ValueQuery, DefaultMigrationStep>;
+	#[pallet::type_value]
+	pub fn DefaultMigrationStep() -> u32 {
+		500
+	}
+
```
Comment (HackFisher) [inline frame/header-mmr/src/lib.rs:116]: LGTM, hard code using 500 is simpler?

File: frame/header-mmr/src/lib.rs [hunk 2]
```diff
@@ -105,10 +105,38 @@ pub mod pallet {
...
 	#[pallet::pallet]
 	pub struct Pallet<T>(_);
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(_n: BlockNumberFor<T>) -> frame_support::weights::Weight {
+			let mut i = 0_u32;
+
+			for _ in
+				frame_support::migration::storage_iter::<u64>(b"DarwiniaHeaderMMR", b"Positions")
+					.drain()
```
Comment (HackFisher) [inline frame/header-mmr/src/lib.rs:126]: This is iterator, and `drain()` will not loop until last elem, right?

**Review by HackFisher — APPROVED** [review]

File: frame/header-mmr/src/lib.rs (hunk 1 above)
Comment (AurevoirXavier) [reply frame/header-mmr/src/lib.rs:116]: I'm not sure if this will affect the chain performance.

File: frame/header-mmr/src/lib.rs (hunk 2 above)
Comment (AurevoirXavier) [reply frame/header-mmr/src/lib.rs:126]: Yes. Iterator are lazy-eval.

Comment (AsceticBear) [reply frame/header-mmr/src/lib.rs:126]: Is there any function like `take()`?

Comment (AurevoirXavier) [reply frame/header-mmr/src/lib.rs:126]: `drain` is `take` for map.

File: frame/header-mmr/src/lib.rs (hunk 1 above)
Comment (HackFisher) [reply frame/header-mmr/src/lib.rs:116]: make sense. We did not have clean the storage right, if we do not set any value.

Comment (AurevoirXavier) [reply frame/header-mmr/src/lib.rs:116]: Yes.

**Review by AsceticBear — APPROVED** [review]

File: frame/header-mmr/src/lib.rs (hunk 1 above)
Comment (AsceticBear) [reply frame/header-mmr/src/lib.rs:116]: I have another about this piece of code. How would you configure it later without a runtime upgrade? I didn't see any setter function. If we can't modify it via signed transaction, has no difference with hard code 500.  @AurevoirXavier

Comment (AurevoirXavier) [reply frame/header-mmr/src/lib.rs:116]: Sudo/Gov

**Pull request merged**